docs/understanding-radicle/how-radicle-works.md

```diff
@@ -0,0 +1,334 @@
+
+There are a few important open-source protocols, concepts, projects, and assets that integrate to make [Radicle's
+vision](https://radicle.mirror.xyz/Qz4Rr0R54KLPurqqswsQ2tfRtBs9MOYpw3dlY9VjyQ8) for decentralized code collaboration
+possible.
+
+- [**Identity**](#radicle-identity): A cryptographically-signed JSON document that identifies a device or a project,
+  stored alongside standard Git data when hosted, cloned, and replicated.
+- [**Network**](#radicle-network): The people who use Radicle tooling and the projects they collaborate on, supported by
+  seed nodes, which provide a data availability layer for projects and identities.
+- [**Peer-to-peer protocol**](#peer-to-peer-protocol): The protocol for disseminating Git repositories via gossip-based
+  replication, enabling the hosting and sharing of projects without knowing where it's physically stored within the
+  Radicle network.
+- [**Command-line tooling (`rad`)**](#command-line-tooling-rad): The core tool for managing your code, project, and
+  identity.
+- [**Web app**](#web-app): The web-based interface for discovering, viewing, and sharing Radicle projects.
+- [**Open-source funding (Drips)**](#open-source-funding-drips): An Ethereum protocol for continuously streaming funds
+  to others, which will be integrated into the [web app](#web-app) as new value flows for developers to sustain their
+  FOSS work.
+
+## Radicle identity
+
+Everything Radicle is trying to build around decentralized code collaboration starts with the concept of identity. In
+Radicle, your identity is a *portable* JSON document that defines a user or a project on the Radicle network.
+
+The most important part is the hashed content identifier—the **peer ID**—which identifies it uniquely wherever it's
+stored or discovered on the Radicle network. Each Radicle identity also has a few required metadata fields, which
+determine whether it's a user or project identity:
+
+```rust
+struct User {
+    /// A short name (nickname, handle), without any prefix such as the `@`
+    /// character
+    name: String,
+}
+
+struct Project {
+    /// A short name
+    name: String,
+
+    /// A slightly longer description (should fit in a headline)
+    description: Option<String>,
+
+    /// The default branch. "master" is assumed for git repositories if
+    /// unspecified.
+    default_branch: Option<String>,
+}
+```
+
+An identity can also be extended with additional metadata as needed by the client or interface that's interacting with
+it.
+
+When you create a user identity, it's signed with an Ed25519 keypair that ensures only you can change the identity and
+push changes to the Radicle network on behalf of your peer ID.
+
+When you create a project, your identity is added as a **delegate**—more on those in a moment.
+
+### Why Radicle starts with identity
+
+At their core, Git repositories are just commit histories. Eventually, these commit histories develop into complex trees
```

Comment:
I'm not sure I agree with this line of reasoning. In the case of a project, you correctly point out that `github.com/bob/project` identifies a project. This is a stable way of referring to a project. The centralised aspect means that GitHub can ensure that there aren't two projects called `project` and so there's no mistaking you're looking at `bob/project` when `bob` shares their project.

So, how is this solved in a decentralised setting? You and I could create `project` and now they're named the same thing. We need a stable identity for our respective `project`s. We look towards [Zooko's triangle](https://en.wikipedia.org/wiki/Zooko%27s_triangle) and have a decentralized identifier (but not human-meaningful :disappointed:). The URN becomes the unique way of identifying a project.

Similarly, for personal identity, we are used to email addresses identifying ourselves. They're also centralised. So what do we do? I create a `Person` that I can share with others that uniquely identifies me as being me.

Note that the uniqueness comes from our Ed25519 keys being unique (in end-of-the-universe-computation theory :smile:).

Reply:
> Identity is stored and shipped within Git data, not a seed node or peer

i.e. all data is local-first

> All actions within a project are cryptographically verified

Yes!

> URNs add uniqueness to naming conventions/schemes that are otherwise centralized

Sure, that implies stable identifiers though :) Uniqueness and searching are two sides of the same coin here

Reply:
Thanks! Gave this section a tweak, mostly to add these items as bullets and remove much of the old content around uniqueness.